skops/io/_sklearn.py

```diff
@@ -171,14 +176,52 @@ def reduce_get_instance(state, src, constructor):
 
 
 def Tree_get_instance(state, src):
-    return reduce_get_instance(state, src, Tree)
+    return reduce_get_instance(state, src, constructor=Tree)
 
 
 def sgd_loss_get_instance(state, src):
     cls = gettype(state)
     if cls not in ALLOWED_SGD_LOSSES:
         raise TypeError(f"Expected LossFunction, got {cls}")
-    return reduce_get_instance(state, src, cls)
+    return reduce_get_instance(state, src, constructor=cls)
+
+
+def reduce_based_on_tuple_get_state(obj, dst):
+    # For more details, take a look at reduce_get_state. This extra function is
+    # needed because __getstate__ of certain sklearn classes returns a tuple
+    # instead of a dict, which we thus need to treat differently.
```

Comment:
why not change reduce so that it can handle a tuple/list?

Reply:
I was thinking about that but was afraid that the function, which is already quite complex, would become too confusing. If you think it's okay, I can work it in.

Reply:
No I think that function deserves to handle this complexity.

Reply:
Is this a "no, don't merge the functions" or a "no, don't keep them separate"? :D

Reply:
lolol, very fair. As in: I think the basic function should handle tuples, so please merge them together and have tuple handling in `reduce_*`.